services/base-service/models/expert/api.js

```diff
@@ -134,17 +134,20 @@ router.route(
     let expertId = model.id + '/' + (pathParts[2] || '');
     let data = req.body;
 
-    logger.info({function:'PATCH'}, JSON.stringify(data));
-
-    if( data.grant ) {
-      const grantRoleModel = await model.get_model('grant_role');
-      await grantRoleModel.patch(data,expertId);
-    } else {
-      const authorshipModel = await model.get_model('authorship');
-      await authorshipModel.patch(data,expertId);
+    try {
+      let resp;
+      let role_model;
+      if( data.grant ) {
+        role_model = await model.get_model('grant_role');
+      } else {
+        role_model = await model.get_model('authorship');
+      }
+      patched=await role_model.patch(data,expertId);
+      res.status(204).json();
+//      res.status(200).json({status: 'ok'});
+    } catch(e) {
+      next(e);
```

Comment:
# Error Handler #
This is the line that catches Model errors and propagates them to the Express Error handler.  We haven't added our own for this API, and just use the Express out of the Box one